packages/tauri-plugin-keygen-rs2/src/machine.rs

```
@@ -27,12 +31,35 @@ pub struct MachineState {
 }
 
 impl MachineState {
+    #[cfg(desktop)]
+    pub fn get_fingerprint() -> String {
+        machine_uid::get().unwrap_or_default()
+    }
+
+    #[cfg(mobile)]
     pub fn get_fingerprint() -> String {
-        machine_uid::get().unwrap_or("".into())
+        "mobile".to_string()
     }
 
-    pub(crate) fn new(app_name: String, app_version: String) -> Self {
-        let fingerprint = Self::get_fingerprint();
+    pub fn get_fingerprint_app<R: Runtime>(app_handle: &AppHandle<R>) -> String {
+        let Some(state) = app_handle.try_machine_uid() else {
+            log::warn!("tauri-plugin-machine-uid is not initialized. You should add it to your application and initialize it before tauri-plugin-keygen-rs2. Using default fingerprint.");
+            return Self::get_fingerprint();
+        };
```

If this fails, then tauri-plugin-machine-uid has not been initialized. I implemented the fallback to avoid a breaking change, but maybe we should bump to 0.5 and require the users to add the missing initialization (and also improve the docs and example).

---

It seems we can keep the original logic unchanged under desktop, and only need to use `tauri-plugin-machine-uid` under mobile. 

And I prefer just let it panic when using mobile and `tauri-plugin-machine-uid` is not initialized, rather than using a default fixed value.

---

I changed the behavior to panic, see my latest comment.